### PR TITLE
AP-3158 add JSON schema validation to irregular incomes

### PR DESCRIPTION
**Pull request by stephenrichards**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-XXX)

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: swagger/v4/swagger.yaml [hunk 1]
```diff
@@ -617,8 +617,6 @@ paths:
       responses:
         '200':
           description: successful
-        '422':
```
Comment (kmahern) [inline swagger/v4/swagger.yaml:620]: I think this won't change if we add the test for unprocessable entity in spec/requests/swagger_docs/v4/irregular_incomes_spec.rb - if we do we would have to rerun the command to regenerate the swagger documentation

File: spec/requests/irregular_incomes_controller_spec.rb [hunk 2]
```diff
@@ -48,7 +48,7 @@
```
Comment (kmahern) [inline spec/requests/irregular_incomes_controller_spec.rb:48]: This spec still contains apipie "show_in_docs" which should be removed

**Comment by kmahern** [comment]
We could also remove references to irregular_incomes from apipie.rb

**Comment by kmahern** [comment]
@stephenrichards some minor comments/suggestions otherwise looks good

File: spec/requests/swagger_docs/v4/irregular_incomes_spec.rb [hunk 3]
```diff
@@ -78,25 +78,25 @@
...
-            ],
-          }
-        end
-
-        run_test! do |response|
-          body = JSON.parse(response.body, symbolize_names: true)
-          expect(body[:errors]).to include(/Missing parameter income_type/)
-        end
-      end
+      # response(422, "Unprocessable Entity") do
```
[the remark this answers is not in the document]
Comment (stephenrichards) [reply spec/requests/swagger_docs/v4/irregular_incomes_spec.rb:81]: Yep, done

**Review by kmahern — APPROVED** [review]

**Pull request merged**